src/spatial/utils.jl

```diff
@@ -266,29 +266,26 @@ end
 
 ############## hopping rates of form L_{s,i,j} ################
 struct HopRatesGeneral{F} <: AbstractHopRates
-    hopping_constants::Vector{Matrix{F}} # hopping_constants[i][j,s] is the hopping constant at site i of species s to jth neighbor
-    rates::Vector{Matrix{F}} # rates[i][j,s] is the hopping rate at site i of species s to jth neighbor
+    hop_const_cumulative_sums::Matrix{Vector{F}} # hop_const_cumulative_sums[s,i] is the vector of cumulative sums of hopping constants of species s at site i
```

Comment:
Why not use an `Array{F,3}` where the first index ranges over the cumulative sums of the neighbors, then species and then location? (Or perhaps location than species if that makes more sense.)

Reply:
Because the number of neighbors is different for different sites

Reply:
Yeah, I was thinking structured grids again…